src/tmpfiles/tmpfiles.c

```diff
@@ -1834,11 +1850,11 @@ static int item_do(Item *i, int fd, fdaction_t action) {
                                 continue;
 
                         de_fd = openat(fd, de->d_name, O_NOFOLLOW|O_CLOEXEC|O_PATH);
-                        if (de_fd >= 0)
-                                /* pass ownership of dirent fd over  */
-                                q = item_do(i, de_fd, action);
+                        if (de_fd < 0)
+                                q = log_error_errno(errno, "Failed to open() file '%s': %m", de->d_name);
                         else
-                                q = -errno;
+                                /* Pass ownership of dirent fd over */
+                                q = item_do(i, de_fd, NULL, action);
```

Comment:
It seems to be the only call site where item_do() is called with path=NULL... wouldn't it be better to build the path here and pass the result to item_do and make the path parameter for all fd_set_*() helper mandatory ?

Reply:
So, I thought about this, and opted against it... I think there's value in passing through user supplied paths as far as possible, but here we are building paths ourselves, and in that case I figured the "normalizing" quality of querying the path of the fd is kinda nice, if you follow what i mean.

The difference is really about "user supplied" and "put together ourselves"

But dunno, I am not convinced too much either way...

Reply:
I see what you meant but the code seems simpler if we make `path` mandatory in the fd_set_*() helpers and it seems better to assume that the caller always knows the path since it provides a fd for it.

The patch makes it optional for just one call site where we actually know the path (even if we need to build it ourselves). And since the point of this patch is to make use of fd_get_path() as less as possible...

Reply:
So I'm inclined to merge this series as is just to keep things moving. I think @fbuihuu's suggestion makes a lot of sense (it'll remove *a* *lot* of duplicate code). @fbuihuu can you later submit a patch that implements your suggestion?